### PR TITLE
[fix-core](block) clear block row_same_bit when block reuse

### LINKED ISSUE #16155: [Bug] row_same_bit DCHECK failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

git commitID: adb758dca

### What's Wrong?

```
F0128 13:24:17.758991 298927 block.h:359] Check failed: row_same_bit.size() == rows() (4366 vs. 307) F0128 13:24:17.761874 298906 block.h:359] Check failed: row_same_bit.
size() == rows() (4996 vs. 937)

Check failure stack trace: ***
@ 0x562c4d8a1359 google::LogMessageFatal::~LogMessageFatal()
@ 0x562c4d8a1359 google::LogMessageFatal::~LogMessageFatal()
@ 0x562c4d8a1359 google::LogMessageFatal::~LogMessageFatal()
@ 0x562c4be928ef doris::vectorized::Block::set_same_bit()
@ 0x562c4be928ef doris::vectorized::Block::set_same_bit()
@ 0x562c4be928ef doris::vectorized::Block::set_same_bit()
@ 0x562c4be8815a doris::vectorized::VMergeIteratorContext::copy_rows()
@ 0x562c4be8815a doris::vectorized::VMergeIteratorContext::copy_rows()
@ 0x562c4be8815a doris::vectorized::VMergeIteratorContext::copy_rows()
@ 0x562c4be9f23b doris::vectorized::VMergeIterator::_next_batch<>()
@ 0x562c4be9f23b doris::vectorized::VMergeIterator::_next_batch<>()
@ 0x562c4be9f23b doris::vectorized::VMergeIterator::_next_batch<>()
@ 0x562c4be94d48 doris::vectorized::VMergeIterator::next_batch()
@ 0x562c4be94d48 doris::vectorized::VMergeIterator::next_batch()
@ 0x562c4be94d48 doris::vectorized::VMergeIterator::next_batch()
@ 0x562c3a4b617b doris::BetaRowsetReader::next_block()
@ 0x562c3a4b617b doris::BetaRowsetReader::next_block()
@ 0x562c3a4b617b doris::BetaRowsetReader::next_block()
@ 0x562c4beb1af6 doris::vectorized::VCollectIterator::Level0Iterator::next()
@ 0x562c4beb1af6 doris::vectorized::VCollectIterator::Level0Iterator::next()
@ 0x562c4beb1af6 doris::vectorized::VCollectIterator::Level0Iterator::next()
@ 0x562c4beb5a2c doris::vectorized::VCollectIterator::Level1Iterator::_normal_next()
@ 0x562c4beb5a2c doris::vectorized::VCollectIterator::Level1Iterator::_normal_next()
@ 0x562c4beb5a2c doris::vectorized::VCollectIterator::Level1Iterator::_normal_next()
@ 0x562c4beb44d4 doris::vectorized::VCollectIterator::Level1Iterator::next()
@ 0x562c4beb44d4 doris::vectorized::VCollectIterator::Level1Iterator::next()
@ 0x562c4beb44d4 doris::vectorized::VCollectIterator::Level1Iterator::next()
@ 0x562c4beaf3f7 doris::vectorized::VCollectIterator::next()
@ 0x562c4beaf3f7 doris::vectorized::VCollectIterator::next()
@ 0x562c4beaf3f7 doris::vectorized::VCollectIterator::next()
@ 0x562c4bec9c35 doris::vectorized::BlockReader::_direct_next_block()
@ 0x562c4bec9c35 doris::vectorized::BlockReader::_direct_next_block()
@ 0x562c4bec9c35 doris::vectorized::BlockReader::_direct_next_block()
@ 0x562c392cbdf0 doris::vectorized::BlockReader::next_block_with_aggregation()
@ 0x562c392cbdf0 doris::vectorized::BlockReader::next_block_with_aggregation()
@ 0x562c392cbdf0 doris::vectorized::BlockReader::next_block_with_aggregation()
@ 0x562c4ca5b9ee doris::vectorized::NewOlapScanner::_get_block_impl()
@ 0x562c4ca5b9ee doris::vectorized::NewOlapScanner::_get_block_impl()
@ 0x562c4ca5b9ee doris::vectorized::NewOlapScanner::_get_block_impl()
@ 0x562c4c731ee4 doris::vectorized::VScanner::get_block()
@ 0x562c4c731ee4 doris::vectorized::VScanner::get_block()
@ 0x562c4c731ee4 doris::vectorized::VScanner::get_block()
@ 0x562c4c707e52 doris::vectorized::ScannerScheduler::_scanner_scan()
@ 0x562c4c707e52 doris::vectorized::ScannerScheduler::_scanner_scan()
@ 0x562c4c707e52 doris::vectorized::ScannerScheduler::_scanner_scan()
@ 0x562c4c70a173 doris::vectorized::ScannerScheduler::_schedule_scanners()::$_2::operator()()
@ 0x562c4c70a173 doris::vectorized::ScannerScheduler::_schedule_scanners()::$_2::operator()()
@ 0x562c4c70a173 doris::vectorized::ScannerScheduler::_schedule_scanners()::$_2::operator()()
@ 0x562c4c70a0c5 std::__invoke_impl<>()
@ 0x562c4c70a0c5 std::__invoke_impl<>()
@ 0x562c4c70a0c5 std::__invoke_impl<>()
0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /mnt/disk1/yuejing/projects/doris/be/src/common/signal_handler.h:428
1# 0x00007FF36B5B1400 in /lib64/libc.so.6
2# __GI_raise in /lib64/libc.so.6
3# __GI_abort in /lib64/libc.so.6
4# 0x0000562C4D8A7329 in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
5# google::LogMessage::SendToLog() in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
6# google::LogMessage::Flush() in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
7# google::LogMessageFatal::~LogMessageFatal() in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
8# doris::vectorized::Block::set_same_bit(std::_Bit_const_iterator, std::_Bit_const_iterator) in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
9# doris::vectorized::VMergeIteratorContext::copy_rows(doris::vectorized::Block*, bool) at /mnt/disk1/yuejing/projects/doris/be/src/vec/olap/vgeneric_iterators.cpp:152
10# doris::Status doris::vectorized::VMergeIterator::_next_batch<doris::vectorized::Block>(doris::vectorized::Block*) at /mnt/disk1/yuejing/projects/doris/be/src/vec/olap
/vgeneric_iterators.h:248
11# doris::vectorized::VMergeIterator::next_batch(doris::vectorized::Block*) at /mnt/disk1/yuejing/projects/doris/be/src/vec/olap/vgeneric_iterators.h:192
12# doris::BetaRowsetReader::next_block(doris::vectorized::Block*) at /mnt/disk1/yuejing/projects/doris/be/src/olap/rowset/beta_rowset_reader.cpp:227
13# doris::vectorized::VCollectIterator::Level0Iterator::next(doris::vectorized::Block*) at /mnt/disk1/yuejing/projects/doris/be/src/vec/olap/vcollect_iterator.cpp:318
14# doris::vectorized::VCollectIterator::Level1Iterator::_normal_next(doris::vectorized::Block*) at /mnt/disk1/yuejing/projects/doris/be/src/vec/olap/vcollect_iterator.cp
p:593
15# doris::vectorized::VCollectIterator::Level1Iterator::next(doris::vectorized::Block*) in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
16# doris::vectorized::VCollectIterator::next(doris::vectorized::Block*) at /mnt/disk1/yuejing/projects/doris/be/src/vec/olap/vcollect_iterator.cpp:208
17# doris::vectorized::BlockReader::_direct_next_block(doris::vectorized::Block*, doris::MemPool*, doris::ObjectPool*, bool*) at /mnt/disk1/yuejing/projects/doris/be/src/
vec/olap/block_reader.cpp:194
18# doris::vectorized::BlockReader::next_block_with_aggregation(doris::vectorized::Block*, doris::MemPool*, doris::ObjectPool*, bool*) at /mnt/disk1/yuejing/projects/dori
s/be/src/vec/olap/block_reader.h:40
19# doris::vectorized::NewOlapScanner::_get_block_impl(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /mnt/disk1/yuejing/projects/doris/be/src/vec/exec/scan/n
ew_olap_scanner.cpp:386
20# doris::vectorized::VScanner::get_block(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /mnt/disk1/yuejing/projects/doris/be/src/vec/exec/scan/vscanner.cpp:
58
21# doris::vectorized::ScannerScheduler::_scanner_scan(doris::vectorized::ScannerScheduler*, doris::vectorized::ScannerContext*, doris::vectorized::VScanner*) at /mnt/dis
k1/yuejing/projects/doris/be/src/vec/exec/scan/scanner_scheduler.cpp:244
22# doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::$_2::operator()() const at /mnt/disk1/yuejing/projects/doris/be/src/vec/e
xec/scan/scanner_scheduler.cpp:168
23# void std::_invoke_impl<void, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::$_2&>(std::_invoke_other, doris::vectorize
d::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::$_2&) at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../.
./../include/c++/11/bits/invoke.h:61
24# std::enable_if<is_invocable_r_v<void, doris::vectorized::ScannerScheduler::schedule_scanners(doris::vectorized::ScannerContext*)::$_2&>, void>::type std::_invoke_r<
void, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::$_2&>(doris::vectorized::ScannerScheduler::_schedule_scanners(doris::ve
ctorized::ScannerContext*)::$_2&) at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:117
25# std::_Function_handler<void (), doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::$_2>::_M_invoke(std::_Any_data const&) at
/mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
26# std::function<void ()>::operator()() const at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_functio
n.h:560
27# doris::PriorityWorkStealingThreadPool::work_thread(int) at /mnt/disk1/yuejing/projects/doris/be/src/util/priority_work_stealing_thread_pool.hpp:134
28# void std::_invoke_impl<void, void (doris::PriorityWorkStealingThreadPool::* const&)(int), doris::PriorityWorkStealingThreadPool*&, int&>(std::_invoke_memfun_deref,
void (doris::PriorityWorkStealingThreadPool::* const&)(int), doris::PriorityWorkStealingThreadPool*&, int&) at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x8
6_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:74
29# std::_invoke_result<void (doris::PriorityWorkStealingThreadPool::* const&)(int), doris::PriorityWorkStealingThreadPool*&, int&>::type std::_invoke<void (doris::Prio
rityWorkStealingThreadPool::* const&)(int), doris::PriorityWorkStealingThreadPool*&, int&>(void (doris::PriorityWorkStealingThreadPool::* const&)(int), doris::PriorityWor
kStealingThreadPool*&, int&) at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96
30# decltype (std::__invoke((this)._M_pmf, std::forward<doris::PriorityWorkStealingThreadPool&>( {parm#1}), std::forward<int&>({parm#1}
))) std::_Mem_fn_base<void (doris:
:PriorityWorkStealingThreadPool::)(int), true>::operator()<doris::PriorityWorkStealingThreadPool&, int&>(doris::PriorityWorkStealingThreadPool*&, int&) const at /mnt/di
sk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/functional:131
31# void std::_invoke_impl<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)>&, doris::PriorityWorkStealingThreadPool&, int&>(std::_invoke_other,
std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)>&, doris::PriorityWorkStealingThreadPool&, int&) at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../l
ib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61
32# std::enable_if<is_invocable_r_v<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)>&, doris::PriorityWorkStealingThreadPool&, int&>, void>::type
std::__invoke_r<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)>&, doris::PriorityWorkStealingThreadPool&, int&>(std::_Mem_fn<void (doris::Prior
ityWorkStealingThreadPool::)(int)>&, doris::PriorityWorkStealingThreadPool&, int&) at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../..
/../../include/c++/11/bits/invoke.h:117
33# void std::Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)> (doris::PriorityWorkStealingThreadPool, int)>::_call<void, , 0ul, 1u
l>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/functional:570
34# void std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)> (doris::PriorityWorkStealingThreadPool, int)>::operator()<>() at /mnt
/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/functional:629
35# void std::__invoke_impl<void, std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)> (doris::PriorityWorkStealingThreadPool, int)
>>(std::__invoke_other, std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)> (doris::PriorityWorkStealingThreadPool, int)>&&) at /m
nt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61
36# std::__invoke_result<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)> (doris::PriorityWorkStealingThreadPool, int)>>::type
std::__invoke<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)> (doris::PriorityWorkStealingThreadPool, int)>>(std::_Bind_result
<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)> (doris::PriorityWorkStealingThreadPool, int)>&&) at /mnt/disk1/yuejing/projects/ldb_toolchain/b
in/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96
37# void std::thread::_Invoker<std::tuple<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityWorkStealingThreadPoo
l*, int)> > >::_M_invoke<0ul>(std::Index_tuple<0ul>) at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std
thread.h:253
38# std::thread::_Invoker<std::tuple<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::)(int)> (doris::PriorityWorkStealingThreadPool, i
nt)> > >::operator()() at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260
39# std::thread::_State_impl<std::thread::_Invoker<std::tuple<std::_Bind_result<void, std::_Mem_fn<void (doris::PriorityWorkStealingThreadPool::*)(int)> (doris::PriorityW
orkStealingThreadPool*, int)> > > >::_M_run() at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h
:211
40# execute_native_thread_routine in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
41# start_thread in /lib64/libpthread.so.0
42# _GI__clone in /lib64/libc.so.6
```

### What You Expected?

Don't crash, return result instead.

### How to Reproduce?

```sql
SELECT coalesce(coalesce(BITMAP_EMPTY(), BITMAP_EMPTY()), CASE
                                                              WHEN ref_12.`cs_bill_cdemo_sk` IS NOT NULL THEN NULL
                                                              ELSE NULL
                                                          END) AS c0
FROM regression_test_datev2_tpcds_sf1_p1.catalog_sales AS ref_12
WHERE BITMAP_SUBSET_IN_RANGE(cast(NULL AS bitmap), cast(BITMAP_MIN(cast(BITMAP_EMPTY() AS bitmap)) AS bigint), cast(ref_12.`cs_bill_addr_sk` AS bigint)) IS NOT NULL
ORDER BY ref_12.`cs_sold_date_sk`
LIMIT 97
OFFSET 125
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by nextdreamblue**
Signed-off-by: nextdreamblue <zxw520blue1@163.com>

# Proposed changes

Issue Number: close #16155

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.27 seconds
 load time: 522 seconds
 storage size: 17120405529 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230128154457_clickbench_pr_85767.html

**Comment by morningman** [comment]
please add test case

**Comment by nextdreamblue** [comment]
> please add test case

done

File: regression-test/suites/datev2/tpcds_sf1_p1/sql/row_same_bit.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+SELECT
```
Comment (zhannngchen) [inline regression-test/suites/datev2/tpcds_sf1_p1/sql/row_same_bit.sql:1]: The SQL query and the issue you fixed seems unrelated, it's not suitable for a test case, we'd better design a simple clear case for the issue, so other developers can understand and modify the case easily.

Comment (nextdreamblue) [reply regression-test/suites/datev2/tpcds_sf1_p1/sql/row_same_bit.sql:1]: ok, i try some clear case, and add some comment for code

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**